src/steps/findings.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  createDirectRelationship,
+  createIntegrationEntity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+  IntegrationMissingKeyError,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../client';
+import { IntegrationConfig } from '../types';
+import { getVulnerabilityLink, getVulnerabilityNumber } from '../util';
+
+export async function fetchFindings({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await apiClient.iterateFindings(async (finding) => {
+    const findingProps = finding.resource;
+    const findingEntity = await jobState.addEntity(
+      createIntegrationEntity({
+        entityData: {
+          source: finding,
+          assign: {
+            _type: 'cobalt_finding',
+            _class: 'Finding',
+            _key: findingProps.id,
+            tag: findingProps.tag,
+            name: findingProps.title,
+            displayName: findingProps.title,
+            description: findingProps.description,
+            category: 'Penetration Testing',
+            typeCategory: findingProps.type_category,
+            labels: JSON.stringify(findingProps.labels, null, 2),
+            impact: JSON.stringify(findingProps.impact, null, 2), //required to be a string in J1 Finding
+            severity: JSON.stringify(findingProps.impact, null, 2), //required property in J1 Finding
+            numericSeverity: findingProps.impact, //required property in J1 Finding
+            likelihood: findingProps.likelihood,
+            state: findingProps.state,
+            open: true, //required property in J1 Finding
+            affectedTargets: JSON.stringify(
+              findingProps.affected_targets,
+              null,
+              2,
+            ),
+            proofOfConcept: findingProps.proof_of_concept,
+            suggestedFix: findingProps.suggested_fix,
+            prerequisites: findingProps.prerequisites,
+            pentestId: findingProps.pentest_id, //value of pentest Assessment _key
+            assetId: findingProps.asset_id, // value of asset _key (which could be class Application or something else)
+            log: JSON.stringify(findingProps.log, null, 2),
```

Comment:
What kind of data is this? Some kinds of content should not be stored in the graph as a property, it can be observed in the raw data view, or the Cobalt UI by providing a `webLink` property on the `Finding`.

Reply:
`log` is a list of `state` transitions with timestamps, according to the schema. I took it out of the graph and left it in raw data. Also, we now have a `webLink`